packages/app/android/src/reactnative/java/io/invertase/firebase/common/SharedUtils.java

```diff
@@ -97,12 +97,10 @@ public static WritableMap getExceptionMap(Exception exception) {
   }
 
   public static String timestampToUTC(long timestamp) {
-    Calendar calendar = Calendar.getInstance();
     long millisTimestamp = timestamp * 1000;
-    Date date = new Date(millisTimestamp + calendar.getTimeZone().getOffset(millisTimestamp));
```

Comment:
Interesting - this method is also used by storage:

```
packages/app/android/src/reactnative/java/io/invertase/firebase/common/SharedUtils.java:  public static String timestampToUTC(long timestamp) {
packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java:            SharedUtils.timestampToUTC(Objects.requireNonNull(tokenResult).getAuthTimestamp()),
packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java:            SharedUtils.timestampToUTC(tokenResult.getExpirationTimestamp()),
packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java:            SharedUtils.timestampToUTC(tokenResult.getIssuedAtTimestamp()),
packages/storage/android/src/main/java/io/invertase/firebase/storage/ReactNativeFirebaseStorageCommon.java:    metadata.putString(KEY_TIME_CREATED, SharedUtils.timestampToUTC(storageMetadata.getCreationTimeMillis() / 1000));
packages/storage/android/src/main/java/io/invertase/firebase/storage/ReactNativeFirebaseStorageCommon.java:    metadata.putString(KEY_UPDATED, SharedUtils.timestampToUTC(storageMetadata.getUpdatedTimeMillis() / 1000));
```

...and this change is semantic, the timestamps *will* be different now for storage.

To be safe the previous behavior might be best preserved, with a new parameter added to the function `boolean useOffset` or similar where storage calls it with `true` and that creates the Calendar instance and does the offset, while Auth does not?

Otherwise the storage code assumptions must be examined to make sure that it can tolerate the new non-local-offset timestamps

Reply:
Thanks!

I think we should be able to get the same time on both iOS and Android, so I investigated the usage in storage.

In storage, it is used to string the creation and update dates of storageReference.
iOS or Android without considering the time zone, I think the behavior will remain the same after this change. The others will be able to get the correct values, but it is a disruptive change.

What do you think?

Reply:
I did not want to "prejudice" your check of the storage code by saying - before you looked at it - that I think essentially the same thing. I think this is more correct even though it is a disruptive change, and I like the change. We could preserve the old behavior but I think the change is actually an improvement. I'll release this in a little bit - I want to see if there is anything else queued up I could get in the release as well.

Thanks!